File: backend-java/src/main/java/com/example/calorie/CalorieApplication.java
package com.example.calorie;

import com.example.calorie.config.JwtProperties;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.EnableConfigurationProperties;

/**
 * カロリー管理アプリ Java バックエンドの起動クラス。
 *
 * <p>このアプリケーションは PostgreSQL のスキーマ所有者であり、
 * Flyway によるマイグレーションを実行する唯一のアプリケーションである。
 * Kotlin 側（ポート 8081）は同じ DB を参照するが、DDL は一切行わない。
 */
@SpringBootApplication
@EnableConfigurationProperties(JwtProperties.class)
public class CalorieApplication {

    public static void main(String[] args) {
        SpringApplication.run(CalorieApplication.class, args);
    }
}

File: backend-kotlin/src/main/kotlin/com/example/calorie/CalorieApplication.kt
package com.example.calorie

import com.example.calorie.config.JwtProperties
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication

/**
 * カロリー管理アプリ Kotlin バックエンドの起動クラス。
 *
 * Java 実装（ポート 8080）と同じ PostgreSQL・同じ API 仕様を共有する。
 * ただし **スキーマの所有者は Java 側だけ** であり、こちらは Flyway を持たない。
 * エンティティとスキーマのずれは `ddl-auto=validate` が起動時に検知する。
 */
@SpringBootApplication
@EnableConfigurationProperties(JwtProperties::class)
class CalorieApplication

fun main(args: Array<String>) {
    runApplication<CalorieApplication>(*args)
}
